### DISCUSSION TITLE
Issues when adding a level

### DISCUSSION THREAD

**Discussion by MarceloDPG** [TrianglyRU/OrbinautFramework#53] [Q&A] [answered]
When i tried adding a level, it has the following issues:

1. Causes a CPU to be created, even when selecting "no player 2" ( if I do select a second player, the 3 CPUs appear, being a CPU of my player character and 2 separate "player 2"s. a bit confusing to explain but the gif shows what I mean)
2. Messes up the Test Level, removes its background,(and depending on where I put the "break;" code) causes the test level and the new level to load at the same time.

If I need to remove the test level, or I'm just being stupid, then let me know.
![HHscpuerror1](https://github.com/user-attachments/assets/f22173e9-9860-44ce-9194-357ca23a2aeb)
![DoubleTrouble](https://github.com/user-attachments/assets/3a1f9dcb-7100-416f-82ef-d4ebd05a032d)

UPDATE: I have since fixed the issues with the level, but not the CPUs

**Comment by TrianglyRU** [comment]
Stage setup is purely code-wise, including CPUs. The game cannot spawn an extra CPU unless you make it to do so, Can I have a look at your stage setup code?

**Reply by MarceloDPG** [reply]
switch room
	{
		case rm_stage_rrz0:
        
		    m_stage_set(99, "RAINFOREST RAMPAGE", 0, bgm_rrz1, [spr_obj_animal_flicky, spr_obj_animal_pocky], room_height, -1, rm_continue, false);    
		    player_spawn(32, 176, global.player_main, "Instances");
			collision_load_binary("widths_s3", "heights_s3", "angles_s3", "CollisionA", "CollisionB");
			
			// Background
			bg_add_layer(spr_rrz_bg_layer0, 0, 120, 0, 0, 0, 0, 0.200, 0);
			bg_set_perspective_x(0.200, 8);
			bg_add_layer(spr_rrz_bg_layer0, 72,  120, 72,  0,  0, 0, 0.225, 0);			// Far Buildings
			bg_add_layer(spr_rrz_bg_layer0, 192, 116, 192, 0,  0, 0, 0.275, 0);			// Far Mechanical Stuff
			bg_add_layer(spr_rrz_bg_layer0, 308, 8, 308, 0, 0, 0, 0.245, 0);			// Bottom Lights
			bg_set_perspective_x(0.315, 2);
			
		break;
	}
I used the example code from the wiki as a base
and copy-pasted some code from the test stage

**Reply by TrianglyRU** [reply]
The code looks fine. If I had to guess based on your former issue, I'd say it is probably related to the way the stage itself is set up, like if c_stage controller was placed twice. Can't say more without looking at the project itself. You can upload it to Google Drive and make it private, and I'll request an access to download it if you wish to keep the project between you and me

**Reply by MarceloDPG** [reply]
I'll tinker around with it some more, and if I can't resolve it, I'll send it your way.

**Reply by MarceloDPG** [reply]
the problem seems to be coming from the player_spawn code.
I deleted it and it only spawns the player character, but that means I cant code where I want sonic to spawn.
It got rid of the duplicates, but it also doesn't allow any CPU characters.

**Reply by TrianglyRU** [reply, marked as answer]
I just tested it with the stock Orbinaut, it works fine. Make sure you do not have obj_player placed in the stage. Having no player_spawn in a stage should result in a crash since objects wouldn't be able to find a player to check for collision with. If your game does not crash, it means the player is spawned by something else

**Reply by MarceloDPG** [reply]
I just realized I've been using the older version of orbinaut 2. im gonna try making a stage in the newer version and see if anything changes.

**Reply by MarceloDPG** [reply]
I just updated and tried it again, and the problem seems to have been fixed.

**Reply by TrianglyRU** [reply]
Shouldn't have been a problem with the 2024.08 release as well. Glad your issue is solved, tho! 👍

**Reply by Otukuoi** [reply]
@MarceloDPG Could you tell/show me exactly what you did to fix said issue? I'm using the newest version of Orbinaut 2 and it's doing the exact same thing you described to me, even though the level I *added* works entirely fine with none of the aforementioned issues.

**Reply by MarceloDPG** [reply]
I went to the newest version and had just copy-pasted the example code from the wiki (chapter 6). I don't know what causes it, or how it fixed that issue, but it did. You could try what I did, and test out a few other versions, but as I said, not too sure what causes it.
